Doc/library/asyncio-task.rst

```diff
@@ -414,6 +414,52 @@ reported by :meth:`asyncio.Task.cancelling`.
    Improved handling of simultaneous internal and external cancellations
    and correct preservation of cancellation counts.
 
+Group Cancellation
+------------------
+
+Cancelling an entire task group is not natively supported by the standard
+library but can be achieved by adding a task to the group that raises an
+exception and ignoring that exception:
+
+.. code-block:: python
+
+   import asyncio
+   from asyncio import TaskGroup
+
+   class CancelTaskGroup(Exception):
+     """Exception raised to cancel a task group."""
+
+   async def job(task_id, sleep_time):
+      print(f'Task {task_id}: start')
+      await asyncio.sleep(sleep_time)
+      print(f'Task {task_id}: done')
+
+   async def cancel_task_group():
+      """A task that would cancel the group it belongs to."""
+      raise CancelTaskGroup()
```

Comment:
I'd move this before 'job', since it's part of the infrastructure everyone needs, while 'job' and 'main' are just examples.

Also change 'would cancel' to 'cancels'.

And possibly rename to 'stop_task_group' since it doesn't use 'CancelledError' or '.cancel()'?